openstudiocore/src/model/Duct.cpp

```diff
@@ -0,0 +1,174 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2015, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#include "Duct.hpp"
+#include "Duct_Impl.hpp"
+
+// TODO: Check the following class names against object getters and setters.
+#include "Connection.hpp"
+#include "Connection_Impl.hpp"
+#include "Connection.hpp"
+#include "Connection_Impl.hpp"
+
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_Duct_FieldEnums.hxx>
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  Duct_Impl::Duct_Impl(const IdfObject& idfObject,
+                       Model_Impl* model,
+                       bool keepHandle)
+    : StraightComponent_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == Duct::iddObjectType());
+  }
+
+  Duct_Impl::Duct_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                       Model_Impl* model,
+                       bool keepHandle)
+    : StraightComponent_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == Duct::iddObjectType());
+  }
+
+  Duct_Impl::Duct_Impl(const Duct_Impl& other,
+                       Model_Impl* model,
+                       bool keepHandle)
+    : StraightComponent_Impl(other,model,keepHandle)
+  {}
+
+  const std::vector<std::string>& Duct_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty()){
+    }
+    return result;
+  }
+
+  IddObjectType Duct_Impl::iddObjectType() const {
+    return Duct::iddObjectType();
+  }
+
+  boost::optional<Connection> Duct_Impl::inletNode() const {
```

Comment:
This should not be here.  You only need to define the [ports](https://github.com/NREL/OpenStudio/blob/df20ac3652c3c3bf1d1172fee3d65434f0780124/openstudiocore/src/model/FanConstantVolume.cpp#L111) and let the base class do the rest of this inlet outlet stuff.